### PR TITLE
Adds bootstrap modal to view partial that displays a legible sanborn legend.

### LINKED ISSUE #795: Expand Sanborn legend by default
From Squirrel's feedback on 8/5/2019.

Add a modal that pops up when you click it that shows the full size legend.

### PR DESCRIPTION
**Pull request by sdellis**
closes #795

### CODE REVIEW

**Comment by tpendragon** [comment]
@sdellis There's a failing test, and can you post a screenshot?

**Comment by sdellis** [comment]
Thanks, @tpendragon. I will fix that test up as I've changed the legend to be open by default. In the meantime, here are a couple screenshots:

default_show
legend_modal

**Comment by sdellis** [comment]
@tpendragon I was able to fix the test. I tried to add a "click" event to the test to make sure the modal works, but it doesn't seem to be finding the id I added as a selector. I wonder if it has something to do with the `render` function?

**Comment by hackartisan** [comment]
@sdellis you might be able to get your test to work if you add `js: true`. example: https://github.com/pulibrary/figgy/blob/a827785c66b4f4117b9692bb68c87fa02a6ef0b7/spec/features/numismatic_ingest_spec.rb#L4

**Comment by sdellis** [comment]
@hackartisan adding `js: true` doesn't seem to help. Perhaps I need to create a feature/capybara test for this?

File: spec/views/catalog/_show_sanborn_legend.erb_spec.rb [hunk 1]
```diff
@@ -2,13 +2,13 @@
 
 require 'rails_helper'
 
-describe 'catalog/_show_sanborn_legend.html.erb', type: :view do
+describe 'catalog/_show_sanborn_legend.html.erb', js: true, type: :view do
```
Comment (hackartisan) [inline spec/views/catalog/_show_sanborn_legend.erb_spec.rb:5]: `js: true` can be removed, right? -- this was put in the feature

File: spec/features/sanborn_legend_spec.rb [hunk 2]
```diff
@@ -5,6 +5,12 @@
...
   it 'renders the card containing the legend for Sanborn Maps' do
     visit solr_document_path('princeton-1r66j405w')
-    expect(page).to have_css('.show-sanborn-legend')
+    expect(page).to have_css('#sanborn-legend')
+  end
+
+  it 'opens the modal containing the legend for Sanborn Maps', js: true do
+    visit solr_document_path('princeton-1r66j405w')
+    click_button('sanborn-legend-button')
+    expect(page).to have_selector('#sanborn-legend-large', visible: true)
```
Comment (hackartisan) [inline spec/features/sanborn_legend_spec.rb:14]: 🎉

**Review by hackartisan — CHANGES_REQUESTED** [review]

File: spec/views/catalog/_show_sanborn_legend.erb_spec.rb (hunk 1 above)
Comment (sdellis) [reply spec/views/catalog/_show_sanborn_legend.erb_spec.rb:5]: Yes, good catch, @hackartisan ! Will fix...

**Review by hackartisan — APPROVED** [review]

**Pull request merged**